types/node.lua

```diff
@@ -204,17 +177,34 @@ function nodetypes.hbox:outputYourself (typesetter, line)
    end
 end
 
-nodetypes.zerohbox = pl.class(nodetypes.hbox)
-nodetypes.zerohbox.type = "zerohbox"
-nodetypes.zerohbox.value = { glyph = 0 }
-
-nodetypes.nnode = pl.class(nodetypes.hbox)
-nodetypes.nnode.type = "nnode"
-nodetypes.nnode.language = ""
-nodetypes.nnode.pal = nil
-nodetypes.nnode.nodes = {}
-
-function nodetypes.nnode:_init (spec)
+--- A zerohbox is a special-kind of hbox with zero width, height and depth.
+--
+-- Derived from `hbox`.
+--
+-- Properties is_zerohbox (and convenience is_zero) and is_box are true.
+-- Note that is_hbox is NOT true: zerohbox are used in a specific context
```

Comment:
"in a specific context" = I wrote this here but I have no idea when a zerohbox is used and should differ from an hbox with 0-dimensions !!!

Reply:
And no specific constructor, so if passed with dimensions etc. it's a zerohbox with dimensions. Maybe a zerohbox is **not** a hbox with zero width, height and depth, after all, but something wholly alien. Who knows what was the intent of the Architect.